mobzy-systems/src/main/kotlin/com/mineinabyss/mobzy/systems/listeners/TamableListener.kt

```diff
@@ -0,0 +1,95 @@
+import com.mineinabyss.mobzy.ecs.components.interaction.Tamable
+import com.mineinabyss.mobzy.ecs.components.interaction.Tamed
+import com.mineinabyss.mobzy.systems.systems.ModelEngineSystem.toModelEntity
+import org.bukkit.Material
+import org.bukkit.Particle
+import org.bukkit.Sound
+import org.bukkit.entity.LivingEntity
+import org.bukkit.event.EventHandler
+import org.bukkit.event.Listener
+import org.bukkit.event.player.PlayerInteractEntityEvent
+import kotlin.random.Random
+
+object TamableListener : Listener {
+
+    /** Tame entities with [Tamable] component on right click */
+    @EventHandler
+    fun PlayerInteractEntityEvent.tameMob() {
+        val mob = (rightClicked as LivingEntity)
+        val gearyEntity = rightClicked.toGearyOrNull() ?: return
+        val maxHealth = gearyEntity.get<MobAttributes>()?.maxHealth ?: 20.0
+        val modelEntity = rightClicked.toModelEntity() ?: return
+        val itemInHand = player.inventory.itemInMainHand
+
+        gearyEntity.with { tamable: Tamable, rideable: Rideable ->
+            when {
+                !gearyEntity.has<Tamed>() -> {
+                    val random = Random(1).nextDouble()
+                    if (tamable.tameItem?.toItemStack() == itemInHand) {
+                        gearyEntity.setPersisting(Tamed)
+                        gearyEntity.get<Tamed>()?.owner = player.uniqueId
+                        player.spawnParticle(
+                            Particle.HEART,
+                            rightClicked.location.apply { y += 1.5 },
+                            10,
+                            random,
+                            random,
+                            random
+                        )
+                    }
+                }
+                else -> {
+                    val tamed = gearyEntity.get<Tamed>() ?: return
+                    when {
+                        tamable.tameItem?.toItemStack() == itemInHand -> {
+                            if (mob.health <= maxHealth) {
+                                if (mob.health + 2 <= maxHealth) mob.health += 2 else mob.health = maxHealth
+                                player.playSound(mob.location, Sound.ENTITY_HORSE_EAT, 1f, 1f)
+                                player.spawnParticle(Particle.HEART, rightClicked.location.apply { y += 2 }, 4)
+                            }
+                            else {
+                                val random = Random(1).nextDouble()
+                                player.spawnParticle(
+                                    Particle.HEART,
+                                    rightClicked.location.apply { y += 1.5 },
+                                    10,
+                                    random,
+                                    random,
+                                    random
+                                )
+                            }
+                        }
+                        tamed.owner == player.uniqueId -> {
```

Comment:
`when` shines when you keep things at one level and dont put one inside the other. If you invert this and do
`tamed.owner != player.uniqueId -> return`, then you can avoid another `when` statement below and keep it at one level.